internal/pkg/service/objectstorage/cloud/azure/storage.go

```diff
@@ -183,6 +183,9 @@ func (s AzureObjectStorageAdapter) CreateBucket(ctx context.Context, bucket *v1a
 					Labels: map[string]string{
 						"giantswarm.io/managed-by": "object-storage-operator",
 					},
+					Finalizers: []string{
+						v1alpha1.BucketFinalizer,
+					},
```

Comment:
When do we delete this finalizer and the secret ?

Reply:
When the secret is deleted here: https://github.com/giantswarm/object-storage-operator/pull/180/files#diff-6cb9e1b1a57fc63376597ad205605e2ef28a47b0059ec5d4c814f9402afb1c60R241

I was wondering if deleting the secret will work if we have a finalizer set on the secret.
Otherwise, I need to patch the secret to remove the finalizer then just after delete the secret.

WDYT ?